src/lib/rules/html-checker/html-checker.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * @fileoverview Validating html using `the Nu html checker`;
+ * https://validator.w3.org/nu/
+ */
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+import { debug as d } from '../../utils/debug';
+import { IRule, IRuleBuilder } from '../../types'; // eslint-disable-line no-unused-vars
+import { RuleContext } from '../../rule-context'; // eslint-disable-line no-unused-vars
+
+import { ITargetFetchEnd, IScanEnd, IProblemLocation, Severity } from '../../types'; // eslint-disable-line no-unused-vars
+
+const debug = d(__filename);
+
+// ------------------------------------------------------------------------------
+// Public
+// ------------------------------------------------------------------------------
+
+const rule: IRuleBuilder = {
+    create(context: RuleContext): IRule {
+        /** The promise that represents the scan by html checker. */
+        let htmlCheckerPromise: Promise<any>;
+        /** The options to pass to the html checker. */
+        const scanOptions = {
+            data: '',
+            format: 'json'
+        };
+
+        type Error = { // eslint-disable-line no-unused-vars
+            extract: string, // code snippet
+            firstColumn: number,
+            lastLine: number,
+            hiliteStart: number,
+            message: string,
+            subType: string
+        };
+
+        const start = (data: ITargetFetchEnd) => {
+            const { response } = data;
+            const htmlChecker = require('html-validator');
```

Comment:
Why is this `require` inside of the function instead of lifted to the top of the scope. It should also be an `import` right?